### PR TITLE
modernize and simplify

### PR DESCRIPTION
**Pull request by grimreaper**
modernize these files

code/src/java/pcgen/core/SettingsHandler.java
code/src/java/pcgen/gui2/facade/Gui2CampaignInfoFactory.java
code/src/java/pcgen/persistence/PersistenceManager.java
code/src/java/pcgen/persistence/SourceFileLoader.java
code/src/java/pcgen/persistence/lst/LstSystemLoader.java
code/src/java/pcgen/persistence/lst/TraitLoader.java
code/src/java/pcgen/persistence/lst/URIEntry.java
code/src/java/pcgen/persistence/lst/URIFactory.java
code/src/java/pcgen/system/FacadeFactory.java
code/src/java/pcgen/util/chooser/ChoiceHandler.java
code/src/java/pcgen/util/chooser/ChooserFactory.java
code/src/java/pcgen/util/chooser/ChooserInterface.java
code/src/java/pcgen/util/chooser/RandomChooser.java
code/src/java/plugin/pretokens/test/PreCampaignTester.java
code/src/test/pcgen/core/prereq/PreCampaignTest.java

and rebuild wrapper to include source

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -166,6 +166,7 @@ dependencies {
     compile group: 'net.sourceforge.argparse4j', name: 'argparse4j', version: '0.7.0'
     compile group: 'net.sourceforge.pcgen', name: 'PCGen-base', version:'1.0.55'
     compile group: 'net.sourceforge.pcgen', name: 'PCGen-Formula', version:'1.0.64'
+    compile group: 'org.jetbrains', name: 'annotations', version:'15.0'
```
Comment (cpmeister) [inline build.gradle]: What is the benefit of the annotations in this library?

File: code/src/java/pcgen/util/chooser/RandomChooser.java [hunk 2]
```diff
@@ -52,33 +51,14 @@
...
-	/**
-	 * Sets the AllowsDups attribute of the Chooser object
-	 *
-	 * @param aBool  The new AllowsDups value
-	 */
-    @Override
-	public void setAllowsDups(boolean aBool)
-	{
-		theAllowDuplicatesFlag = aBool;
-	}
```
Comment (cpmeister) [inline code/src/java/pcgen/util/chooser/RandomChooser.java]: I'm pretty sure that this setter has a purpose, even if it isn't used. Please don't remove it.

File: build.gradle (hunk 1 above)
Comment (grimreaper) [inline build.gradle]: It serves as a hinter for FindBugs, Eclipse/IntelliJ, and various other tools.
See for example http://help.eclipse.org/kepler/index.jsp?topic=/org.eclipse.jdt.doc.user/tasks/task-using_null_annotations.htm

**Comment by cpmeister** [comment]
I think that adding final to parameters that don't need it shouldn't be encouraged to final nor enforced. I can understand from a coding standpoint for new code to have final parameters at the coder's discretion but I don't think it is something that existing code should be changed because of.

File: build.gradle (hunk 1 above)
Comment (cpmeister) [inline build.gradle]: Is it possible to use JSR 305 annotations instead? Netbeans (the IDE I use) doesn't support the jetbrains annotations as far as I know.

Comment (grimreaper) [inline build.gradle]: Its the same thing AFAIK. You just need a place to get the annotations from.
You can also get the same annotations from com.google.code.findbugs

**Comment by grimreaper** [comment]
moving the annotations into a separate PR.  It is going to be a base commit for future work on PlayerCharacter.

**Pull request closed without merging**